### DISCUSSION TITLE
Problem bei der Installation von FroshPlatformThumbnailProcessor – Abhängigkeitskonflikte

### DISCUSSION THREAD

**Discussion by yyaylak** [FriendsOfShopware/FroshPlatformThumbnailProcessor#146] [Q&A]
### Problem beim Update von `FroshPlatformThumbnailProcessor` über Plugin-Manager – Abhängigkeitskonflikte

**Beschreibung:**

Ich habe versucht, das Plugin `frosh/platform-thumbnail-processor` (Version 5.2.1) über den Shopware Plugin-Manager zu aktualisieren, bin jedoch auf Probleme mit der Abhängigkeitsauflösung gestoßen. Es gibt Konflikte zwischen dem Plugin und der in meiner Installation vorhandenen Version von `shopware/core`, sowie dem Plugin `dreisc-seo-pro`.

**Fehlermeldung:**

```bash
Could not execute "composer require" for plugin 
"FroshPlatformThumbnailProcessor (frosh/platform-thumbnail-processor:5.2.1).
Output: ./composer.json has been updated 
Running composer update frosh/platform-thumbnail-processor --with-dependencies
> pre-update-cmd: Symfony\Flex\Flex_composer_tmp1->configureInstaller 
Loading composer repositories with package information
Dependency shopware/core is also a root requirement. Package has not been 
listed as an update argument, so keeping locked at old version. 
Use --with-all-dependencies (-W) to include root dependencies.
> pre-pool-create: Symfony\Flex\Flex_composer_tmp1->truncatePackages
Updating dependencies Dependency resolution completed in 0.001 seconds
Your requirements could not be resolved to an installable set of packages.

Problem 1: 
- Root composer.json requires dreisc/dreisc-seo-pro 6.12.0 
  (exact version match: 6.12.0 or 6.12.0.0), 
  found dreisc/dreisc-seo-pro[6.12.3] but it does not match your constraint 
  and is therefore not installable.
Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

**Details:**

- Shopware Version: 6.6.5.1
- Plugin: FroshPlatformThumbnailProcessor (Version 5.2.1)
- PHP Version:  8.2.23

**Probleme:**
1. Das Update des Plugins über den Plugin-Manager schlägt fehl, da die Abhängigkeit `shopware/core` bei der alten Version gesperrt bleibt. Der Fehler deutet darauf hin, dass möglicherweise eine neuere Version benötigt wird, jedoch bleibt diese gesperrt.
2. Das Plugin `dreisc-seo-pro` ist auf Version `6.12.0` in meiner `composer.json` festgelegt, es gibt jedoch eine neuere Version `6.12.3`, die nicht kompatibel ist.

**Schritte zur Reproduktion:**

1. Gehe in den Plugin-Manager.
2. Versuche, das Plugin `FroshPlatformThumbnailProcessor` auf Version 5.2.1 zu aktualisieren.
3. Die Installation schlägt fehl und zeigt die oben genannte Fehlermeldung.

**Erwartetes Verhalten:**

Das Plugin sollte erfolgreich aktualisiert werden können, ohne dass es zu Konflikten mit `shopware/core` und `dreisc-seo-pro` kommt.

**Zusätzliche Informationen:**

Das Update wurde über den Plugin-Manager und **nicht** über Composer durchgeführt. Es wäre hilfreich, wenn die Abhängigkeiten des Plugins auf Kompatibilitätsprobleme überprüft werden könnten.

Vielen Dank für die Unterstützung!

**Comment by tinect** [comment]
Der Fehler deutet darauf hin, dass es nichts mit diesem Plugin zu tun hat.
Auch kann der Meldung kein Problem mit "shopware/core" entnehmen.
Der Fehler tritt vermutlich auch auf, wenn du irgendein anderes Plugin aktualisierst oder installierst.

In deiner composer.json steht die falsche Version vom dreisc-seo-pro plugin. Vermutlich wurde entweder die composer.json manipuliert oder das Plugin auf dem Speicher. Bitte korrigiere das.

**Reply by yyaylak** [reply]
> Der Fehler deutet darauf hin, dass es nichts mit diesem Plugin zu tun hat. Auch kann der Meldung kein Problem mit "shopware/core" entnehmen. Der Fehler tritt vermutlich auch auf, wenn du irgendein anderes Plugin aktualisierst oder installierst.
> 
> In deiner composer.json steht die falsche Version vom dreisc-seo-pro plugin. Vermutlich wurde entweder die composer.json manipuliert oder das Plugin auf dem Speicher. Bitte korrigiere das.

Vielen Dank für den Hinweis, Tinect! Andere Plugins lassen sich problemlos updaten oder installieren, das funktioniert. Ich schaue mir das Thema mit Composer gerne mal genauer an.

**Comment by Elements-Webdesign** [comment]
Hallo zusammen,

ich versuche, das Plugin frosh/platform-thumbnail-processor (Version 5.2.1) in meine Shopware-Installation zu integrieren. Leider scheitert der Befehl composer require aufgrund eines Abhängigkeitskonflikts.

```
Could not execute "composer require" for plugin "FroshPlatformThumbnailProcessor (frosh/platform-thumbnail-processor:5.2.1).
...
Problem 1
- Root composer.json requires frosh/platform-thumbnail-processor 5.2.1 -> satisfiable by frosh/platform-thumbnail-processor[5.2.1].
- frosh/platform-thumbnail-processor 5.2.1 requires nikic/php-parser ^4.13.0 but the package is fixed to v5.4.0 (lock file version).
```

Die Fehlermeldung deutet darauf hin, dass frosh/platform-thumbnail-processor eine ältere Version von nikic/php-parser benötigt (^4.13.0). In meinem Projekt ist jedoch bereits die Version v5.4.0 installiert, die durch eine frühere Composer-Aktion gesperrt ist.

Was ich bereits probiert habe:

Installation mit --with-all-dependencies:

`composer require frosh/platform-thumbnail-processor:5.2.1 --with-all-dependencies`

Leider ohne Erfolg – es bleibt beim Konflikt.

Installation einer älteren Version von nikic/php-parser:

`composer require nikic/php-parser:^4.19 --with-all-dependencies
`

Shopware-Version: 6.6.9.0
Danke für Euren Input.

LG
Johannes

**Reply by tinect** [reply]
Brauchst du denn das Package in der Version? Das wird aus deinem Text leider nicht klar. Wenn nicht, entferne es, andernfalls bitte ein Issue aufmachen, damit sich jemand das angucken  kann.

**Reply by Elements-Webdesign** [reply]
Guten Morgen,
vielen Dank für Deine Antwort @tinect .

Ja laut Composer benötigen folgende Projekte den Parser ab Version 5.X

```
shopware/production       -      requires nikic/php-parser (^5.4)
phpunit/php-code-coverage 11.0.8 requires nikic/php-parser (^5.3.1)
sebastian/complexity      4.0.1  requires nikic/php-parser (^5.0)
sebastian/lines-of-code   3.0.1  requires nikic/php-parser (^5.0)
```

Ich werde dann mal ein Issue aufmachen.

Lieben Dank
